load-generator/src/main/java/com/amazon/opentelemetry/load/generator/emitter/OtlpMetricEmitter.java

```diff
@@ -47,43 +53,46 @@ public void emitDataLoad() throws Exception {
 
   @Override
   public void nextDataPoint() {
-    apiLatencyRecorder.record(200, Labels
-        .of(DIMENSION_API_NAME, API_NAME, DIMENSION_STATUS_CODE, "200"));
-
     apiBytesSentCounter
-        .add(100, Labels.of(DIMENSION_API_NAME, API_NAME, DIMENSION_STATUS_CODE, "200"));
+            .add(10, METRIC_ATTRIBUTES);
+    mutateApiLatency();
   }
 
   @Override
   public void setupProvider() {
-    MetricExporter metricExporter =
-        OtlpGrpcMetricExporter.builder()
-            .setChannel(
-                ManagedChannelBuilder.forTarget(param.getEndpoint()).usePlaintext().build())
-            .build();
-
-    IntervalMetricReader.builder()
-        .setMetricProducers(
-            Collections.singleton(SdkMeterProvider.builder().buildAndRegisterGlobal()))
-        .setExportIntervalMillis(param.getFlushInterval())
-        .setMetricExporter(metricExporter)
-        .build();
-
-    Meter meter = GlobalMeterProvider.getMeter("aws-otel-load-generator-metric", "0.1.0");
-
-
-    apiBytesSentCounter = meter
-        .longCounterBuilder(API_COUNTER_METRIC)
-        .setDescription("API request load sent in bytes")
-        .setUnit("one")
-        .build();
-
-    apiLatencyRecorder =
-        meter
-            .longValueRecorderBuilder(API_LATENCY_METRIC)
+    Resource resource = Resource.getDefault().merge(
+            Resource.create(Attributes.of(ResourceAttributes.SERVICE_NAME, "load-generator")));
+
+    OpenTelemetrySdk.builder().setMeterProvider(
+                    SdkMeterProvider.builder()
+                            .registerMetricReader(
+                                    PeriodicMetricReader.builder(OtlpGrpcMetricExporter.builder().setEndpoint("http://" + param.getEndpoint()).build())
+                                            .setInterval(Duration.ofMillis(param.getFlushInterval())).build())
+                            .setResource(resource)
+
+                            .build())
+            .buildAndRegisterGlobal();
+
+    Meter meter =
+            GlobalOpenTelemetry.meterBuilder("aws-otel-load-generator-metric")
+                    .setInstrumentationVersion("1.0")
+                    .build();
+
+    apiBytesSentCounter =
+            meter.counterBuilder(API_COUNTER_METRIC)
+                    .setDescription("API request load sent in bytes")
+                    .setUnit("one")
+                    .build();
+
+    meter.gaugeBuilder(API_LATENCY_METRIC)
```

Comment:
Maybe you can use a histogram to record latency instead of a gauge: https://aws-otel.github.io/docs/getting-started/java-sdk/trace-manual-instr#creating-metrics

Reply:
thanks. as discussed keeping Gauge for now.